Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -415,16 +360,111 @@ public Task<object> InvokeCommandAsync(LSP.ExecuteCommandParams request, Cancell
             await _rpcWrapper.NotifyWithParameterObjectAsync(request, parameters).ConfigureAwait(false);
         }
 
+        private LanguageServerSettings.PythonSettings GetSettings(Uri scopeUri = null)
+        {
+            IPythonLanguageClientContext context = null;
+            if (scopeUri == null) {
+                // REPL context has null RootPath
+                context = _clientContexts.Find(c => c.RootPath == null);
+                if (context == null) {                  
+                    return null;
+                }
+            }else {
+                var pathFromScopeUri = CommonUtils.NormalizeDirectoryPath(scopeUri.LocalPath).ToLower();
+                // Find the matching context for the item
+                context = _clientContexts.Find(c => scopeUri != null && PathUtils.IsSamePath(c.RootPath.ToLower(), pathFromScopeUri));
```

Comment:
finding the correct context was the source of a lot of problems, some paths had an extra '\' at the end